File: backend/src/main/java/com/j_mikolajczyk/momentum/models/Week.java
package main.java.com.j_mikolajczyk.momentum.models;
import java.util.List;

public class Week {
    private List<Day> days;

    public Week() {
    }

    public Week(List<Day> days) {
        this.days = days;
    }

    public List<Day> getDays() {
        return this.days;
    }

    public void setDays(List<Day> days) {
        this.days = days;
    }
    
}

File: frontend/src/models/Week.js
export default class Week {

    constructor(days = []) {
        this.days = days.map(day => ({
            ...day,
            exercises: day.exercises.map(exercise => ({
                ...exercise,
                sets: exercise.sets.map(() => ({ weight: '', reps: '', logged: false }))
            }))
        }));
    }

    getDays() {
        return this.days;
    }

    setDays(days) {
        this.days = days;
    }
}